### ISSUE TITLE
Login is not working

### ISSUE THREAD

**Issue by promentol** [scholtzm/meteor-accounts-steam#4]
I have configured the system using ServiceConfiguration, and I am using redirect approach
After successful login in steam, at the time of getting redirected to my app, no user is being created.
On server side I have added listener Accounts.onLoginFailure , here is the error I am getting "No matching login attempt found"
I am using Meteor 1.4.1.1

**Comment by scholtzm** [comment]
I'll test it out today with latest Meteor and I'll let you know.

**Comment by scholtzm** [comment]
Works as expected, running Meteor 1.4.1.1.

Here's my service configuration:

``` js
  ServiceConfiguration.configurations.upsert(
    { service: 'steam' },
    {
      $set: {
        loginStyle: 'redirect',
        timeout: 5000
      }
    }
  );
```

I'm also using:

``` js
Accounts.onCreateUser
Accounts.onLogin
Accounts.validateLoginAttempt
```

but these are not required for the package to work.

**Comment by sebmorris** [comment]
I am getting the same error. "No matching login attempt found"

I have this on the server:

```
Meteor.startup(() => {
     ServiceConfiguration.configurations.upsert(
          { service: 'steam' },
          {
               $set: {
                    loginStyle: 'redirect',
                    timeout: 5000
               }
          }
     );
});
```

In client side html:

```
<body>
     {{> login}}
     {{> loginButtons}}
</body>

<template name="login">
     <button id="login-button">Steam login</button>
</template>
```

Server side JS: 

```
Template.login.events({
     'click #login-button': function() {
          console.log('In the event');
          console.log(Meteor.loginWithSteam);
          Meteor.loginWithSteam({
               loginStyle: 'redirect'
          }, function(e) {
               throw new Meteor.Error(e);
          });
     }
});
```

As you can see I have tried both the `{{> loginButtons}}` method and the `Meteor.loginWithSteam` function, neither works with my current code.
I have removed most of the packages (in an attempt to work out what I was missing), and only have (beyond core packages):

```
scholtzm:accounts-steam
service-configuration
random
accounts-ui
```

Any insight would be very much appreciated.

**Comment by scholtzm** [comment]
This is weird but I'm glad you chipped in with your report. I'll create a minimal example soon, hopefully even today.

One thing I can tell you though is that that `loginWithSteam` has no callback. You are getting redirected immediatelly and therefore there is no callback once you get logged in.

I call it like this:

``` js
Template.navigation.events({
  'click #login-button': function() {
    Meteor.loginWithSteam();
  }
});
```

You can use [`Accounts.onLogin` (client)](https://docs.meteor.com/api/accounts-multi.html#AccountsCommon-onLogin) for this.

**Comment by scholtzm** [comment]
I can confirm that this is indeed a bug. The weird thing is, my gradually updated Meteor app [works just fine](https://github.com/scholtzm/meteor-accounts-steam/issues/4#issuecomment-251767078).

I'm looking into it.

**Comment by sebmorris** [comment]
Great, thanks very much :).

**Comment by scholtzm** [comment]
Ok, I figured out what the issue was. I'll be pushing the update ASAP.

**Comment by sebmorris** [comment]
Cool, thanks. Do you mind saying what it was? Ill check again after the update has been pushed.

**Comment by scholtzm** [comment]
I just published v1.2.0 and created this [new basic example](https://github.com/scholtzm/meteor-accounts-steam-example).

I also updated the README - login style `popup` never really worked actually. OpenID doesn't support it AFAIK.

The main issue was caused by [locking dependency versions in package.js](https://github.com/scholtzm/meteor-steam/commit/1c3d877005374ae30e6218d1316c2c18903f96a7). Removing them solved the problems.

I'm still perplexed why my old app worked and new apps did not. I tried digging in the code but Meteor's monorepo didn't make it very easy to track changes. :-)

Let me know if the fix works for you and we can close this.
